src/packet/bwe/arrival_group.rs

```diff
@@ -0,0 +1,309 @@
+use std::mem;
+use std::time::{Duration, Instant};
+
+use crate::rtp::SeqNo;
+
+use super::AckedPacket;
+
+const BURST_TIME_INTERVAL: Duration = Duration::from_millis(5);
+
+#[derive(Debug, Default)]
+pub struct ArrivalGroup {
+    first_seq_no: Option<SeqNo>,
+    last_seq_no: Option<SeqNo>,
+    first_local_send_time: Option<Instant>,
+    local_send_time: Option<Instant>,
+    remote_recv_time: Option<Instant>,
+    size: usize,
+}
+
+impl ArrivalGroup {
+    /// Maybe add a packet to the group.
+    ///
+    /// ## Return
+    ///
+    /// * [`Belongs::Belongs`] if the packet belongs to the group and was added.
+    /// * [`Belongs::NewGroup`] if the packet is the start of a new group and thus wasn't added.
+    ///                         The caller should create a new group and add this packet to the
+    ///                         group.
+    /// * [`Belongs::Skipped`] if the packet was skipped i.e. because it was out of order.
+    fn add_packet(&mut self, packet: AckedPacket) -> Belongs {
+        let belongs = self.belongs_to_group(&packet);
+
+        if belongs == Belongs::NewGroup || belongs == Belongs::Skipped {
+            return belongs;
+        }
+
+        self.remote_recv_time = Some(packet.remote_recv_time);
+
+        if self.first_local_send_time.is_none() {
+            self.first_local_send_time = Some(packet.local_send_time);
+            self.first_seq_no = Some(packet.seq_no);
+        }
+
+        self.local_send_time = self.local_send_time.max(Some(packet.local_send_time));
+        self.size += 1;
+        self.last_seq_no = Some(packet.seq_no);
```

Comment:
Is true if packets are added out of order?

Reply:
It wouldn't be, can it happen though? This derives from TWCC reports which are always ordered, I guess individual reports could arrive out of order though

In any case this and first_seq_no are only used for logging, still worth fixing.